### DISCUSSION TITLE
Types on GetInputProps type?

### DISCUSSION THREAD

**Discussion by Zachary-Kaye** [mantinedev/mantine#3819] [Q&A]
I have just begun using Mantine, following the recipes for [https://mantine.dev/form/recipes/](https://mantine.dev/form/recipes/). The source code for the examples uses the spread operator when getting input props. However, when using Mantine in a Next.js/Typescript project, I have found that I can not use the spread operator as in the example due to the following error: `Type ...{ value: any; onChange: any; checked?: any; error?: any; onFocus?: any; onBlur?: any } is not assignable to Type ReactNode | undefined`  

The type of ReactNode is defined as `ReactElement | string | number | ReactFragment | ReactPortal | boolean | null | undefined`, whereas the result of GetInputProps is `{ value: any; onChange: any; checked?: any; error?: any; onFocus?: any; onBlur?: any }`. Since the return types of GetInputProps are all any, when spreading it is unable to assign these to type ReactNode? Would updating the types of GetInputProps to be more specific fix this issue, or would there be implications elsewhere? For example, what types can value be as GetInputProps is used by multiple different input fields? In the meantime, I could individually assign the fields, instead of spreading them. However, this does not seem like a long-term solution as it gets repetitive fast. Please let me know if my understanding is correct, as I still have much to learn about Mantine and Typescript. If anyone also has any other alternatives, please reach out :)

**Comment by rtivital** [comment]
Values are all any to provide ts compatibility with all kind of inputs. It may break if other types are used (or require significant efforts to maintain). Can you send a sandbox that reproduces the issue that you have? Template – https://codesandbox.io/s/mantine-template-9ze89?file=/src/App.tsx:0-6

**Reply by Zachary-Kaye** [reply]
Hey, from further investigation, it appears the issue was with my IDE. I tried replicating the problem in the sandbox with the same project setup, but I was unsuccessful. Therefore, I ended up playing around with my IDE (Jetbrains Webstorm), and I am no longer getting this error. In fact, it was similar in nature to this [https://stackoverflow.com/questions/73604786/rxjs-typescript-and-the-dreaded-type-mismatch-warning-error](https://stackoverflow.com/questions/73604786/rxjs-typescript-and-the-dreaded-type-mismatch-warning-error). Thanks :)
